internal/metrics/metrics.go

```diff
@@ -0,0 +1,208 @@
+package metrics
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/base-org/pessimism/internal/core"
+	"github.com/base-org/pessimism/internal/logging"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/collectors"
+
+	"go.uber.org/zap"
+)
+
+const (
+	metricsNamespace    = "pessimism"
+	SubsystemInvariants = "invariants"
+	SubsystemEtl        = "etl"
+)
+
+const serverShutdownTimeout = 10 * time.Second
+
+type Config struct {
+	Host              string
+	Port              int
+	Enabled           bool
+	ReadHeaderTimeout int
+}
+
+type Metricer interface {
+	IncActiveInvariants()
+	DecActiveInvariants()
+	IncActivePipelines()
+	DecActivePipelines()
+	RecordInvariantRun(invariant string)
+	RecordAlertGenerated(invariant string)
+	RecordNodeError(node string)
+	RecordUp()
+	Start()
+	Shutdown(ctx context.Context) error
+	Document() []DocumentedMetric
+}
+
+type Metrics struct {
+	ActiveInvariants prometheus.Gauge
+	ActivePipelines  prometheus.Gauge
+	Up               prometheus.Gauge
+	InvariantRuns    *prometheus.CounterVec
+	AlarmsGenerated  *prometheus.CounterVec
+	NodeErrors       *prometheus.CounterVec
+
+	registry *prometheus.Registry
+	factory  Factory
+	server   *http.Server
+}
+
+var Stats Metricer = new(noopMetricer)
```

Comment:
Is this used anywhere? Curious to know why we're initializing two variables here